tests/attacks/inference/test_membership_inference.py

```diff
@@ -0,0 +1,363 @@
+        assert (train_pos > test_pos or
+                train_pos == pytest.approx(test_pos, abs=0.03) or
+                test_pos == 1)
+
+
+def test_rule_based_tabular(get_iris_dataset, get_tabular_classifier_list):
+    classifier_list = get_tabular_classifier_list(MembershipInferenceBlackBoxRuleBased)
+    if not classifier_list:
+        logging.warning("Couldn't perform this test because no classifier is defined")
+        return
+
+    (x_train, y_train), (x_test, y_test) = get_iris_dataset
+
+    for classifier in classifier_list:
+        # print(type(classifier).__name__)
+        attack = MembershipInferenceBlackBoxRuleBased(classifier)
+        # infer attacked feature
+        inferred_train = attack.infer(x_train, y_train)
+        inferred_test = attack.infer(x_test, y_test)
+        # check accuracy
+        train_pos = sum(inferred_train) / len(inferred_train)
+        test_pos = sum(inferred_test) / len(inferred_test)
+        assert (train_pos > test_pos or
+                train_pos == pytest.approx(test_pos, abs=0.06) or
+                test_pos == 1)
+
+
+def test_black_box_image(fix_get_mnist_subset, get_image_classifier_list_for_attack):
+    classifier_list = get_image_classifier_list_for_attack(MembershipInferenceBlackBox)
+    if not classifier_list:
+        logging.warning("Couldn't perform this test because no classifier is defined")
+        return
+
+    x_train, y_train, x_test, y_test = fix_get_mnist_subset
+    attack_train_size = int(len(x_train) * attack_train_ratio)
+    attack_test_size = int(len(x_test) * attack_train_ratio)
+
+    for classifier in classifier_list:
+        attack = MembershipInferenceBlackBox(classifier)
+        # train attack model using only attack_train_ratio of data
+        attack.fit(x_train[:attack_train_size], y_train[:attack_train_size],
+                   x_test[:attack_test_size], y_test[:attack_test_size])
+        # infer attacked feature on remainder of data
+        inferred_train = attack.infer(x_train[attack_train_size:], y_train[attack_train_size:])
+        inferred_test = attack.infer(x_test[attack_test_size:], y_test[attack_test_size:])
+        # check accuracy
+        train_pos = sum(inferred_train) / len(inferred_train)
+        test_pos = sum(inferred_test) / len(inferred_test)
+        assert (train_pos > test_pos or
+                train_pos == pytest.approx(test_pos, abs=0.03) or
+                test_pos == 1)
+
+
+def test_black_box_tabular(get_tabular_classifier_list, get_iris_dataset):
+    classifier_list = get_tabular_classifier_list(MembershipInferenceBlackBox)
+    if not classifier_list:
+        logging.warning("Couldn't perform this test because no classifier is defined")
+        return
+
+    (x_train, y_train), (x_test, y_test) = get_iris_dataset
+    attack_train_size = int(len(x_train) * attack_train_ratio)
+    attack_test_size = int(len(x_test) * attack_train_ratio)
+
+    model_types = ['nn', 'rf', 'gb']
```

Comment:
the `model_types` parameters should be set as a pytest marker parameter. See `@pytest.mark.parametrize("norm", [np.inf, 1, 2])` in test `tests.attacks.evasion.test_fast-gradient.test_norm_images`